do-not-commit/no-auth-over-http.py

```diff
@@ -0,0 +1,52 @@
+import requests
+
+# ok
+good_url = "https://www.github.com"
+bad_url = "http://www.github.com"
+
+# ruleid:no-auth-over-http
+r = requests.post("http://www.github.com", auth=("user", "pass"))
+
+# ok
+r = requests.post(good_url, auth=("user", "pass"))
+
+# ok
+r = requests.get(bad_url, timeout=50)
+
+
+def test1():
+    # ruleid:no-auth-over-http
+    bad_url = "http://www.github.com"
```

Comment:
Authentication detected over HTTP. HTTP does not provide any
encryption or protection for these authentication credentials.
This may expose these credentials to unauthhorized parties.
Use 'https://' instead. ([y.no-auth-over-http.no-auth-over-http](https://semgrep.dev/r/y.no-auth-over-http.no-auth-over-http))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=y.no-auth-over-http.no-auth-over-http) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>